42225233 lookup resource by labels

ke4 opened the pull request:
Added new tests related to search by labels to search_plate spec.
Also created search related specs for tubes and flowcells.

Commit:
[#41182989] Init commit for embedded label for tube

Commit:
[#41182989] Refactored embedded label for tube

The specs are still failing for 
the other resources!!!

maxigit committed:
Fix session bug

maxigit committed:
Use Labellable#labels_to_stream

Commit:
manual merge - updating development to latest version

Commit:
fixed the problem caused the prev. manual merge

Commit:
[#42225233] added search by labels to plate search

Commit:
[#42225233] Removed unnecessary shared context

and set the labels with the plate's uuid

Commit:
[#42225233] Moved assets related stuff to a shared file

Commit:
[#42225233] Continue 'DRY'ing the spec for tube, flowcell

Commit:
[#42225233] Added search specs for tube and flowcell

Commit:
[#42225233] Removed unnecessary shared context

It has been added by accident.

maxigit commented on spec/integrations/search_shared.rb:
```diff
@@ -0,0 +1,123 @@
+  let(:asset_type) { "resource" }
+  let(:label_position) { "front barcode" }
+  let(:label_type) { "sanger-barcode" }
+  let(:labels) {
```
I'm a bit confused, is labels a list of Labels as its name suggests or a hash of uuid ?

Shouldn't they be barcode anyway ?

ke4 replied:
Actually, we don't need them at all.
Earlier, this was a list of strings like "barcode for plate_#{i}", but I changed it to test something for Loic with uuids and later I left the uuids there.
But I can use the asset_uuid array for the same purpose.

So, I will remove the 'labels' from the search_shared file and use the 'asset_uuid' instead of it.

maxigit commented on spec/integrations/search_shared.rb:
```diff
@@ -0,0 +1,123 @@
+      (0..9).map do |i|
+        session << labellable = Lims::Core::Laboratory::Labellable.new(:name => asset_uuids[i],
+                                                           :type => asset_type)
+        set_uuid(session, labellable, labellable_uuids[i-1])
```
you can use the zip function to iterate on 2 enumerable at the same time

(0..9).zip(labellable_uuids) do |i, labellables_uuid|

However in that particular case, labellables_uuids.each_with_index seems more appropriate. Am i wrong ?

ke4 replied:
I use the '(0..9).map do |i|', because I need the i variable in the following 2 cases:
- asset_uuids[i]
- labellable_uuids[i-1] -> it should be labellable_uuids[i](I will correct it)
  asset_uuid and labellable_uuid is not the same.
  So, I think, your suggestion is not good.

Commit:
[#42225233] Removed unnecessary defined hash

Commit:
[#42225233] Refactoring the spec

I'm using the 'zip' function to iterate over 2 array,
instead of '(0..9).map do |i| ...'

maxigit commented:
Could you rebase it please.
